lib/worker/batcher.ex

```diff
@@ -655,17 +655,22 @@ defmodule BorsNG.Worker.Batcher do
   end
 
   defp put_incomplete_on_hold(repo_conn, batch) do
-    batches_query = batch.project_id
+    batches = batch.project_id
     |> Batch.all_for_project(:running)
     |> where([b], b.id != ^batch.id and b.priority < ^batch.priority)
+    |> Repo.all()
+
+    ids = Enum.map(batches, &(&1.id))
+
     Status
-    |> join(:inner, [s], b in ^batches_query, s.batch_id == b.id)
+    |> where([s], s.batch_id in ^ids)
     |> Repo.delete_all()
-    batches_query
-    |> Repo.all()
-    |> Enum.each(&send_status(repo_conn, &1, :delayed))
-    Repo.update_all(batches_query,
-      set: [state: :waiting])
+
+    
+    Enum.each(batches, &send_status(repo_conn, &1, :delayed))
+    Batch
+    |> where([b], b.id in ^ids)
+    |> Repo.update_all(set: [state: :waiting])
```

Comment:
I practice, the changes to this function, if affecting DB performance, will improve it, because we were going to fetch the batches before, and by using the `ids` directly instead of the more complex query for join might benefit the query planner (unless it's properly caching and identifying that the resultset would be the same, in such case there's virtually no difference between both).

So, this change does not degrade performance nor readability :>

Reply:
Agreed.